src/main/scala/is/hail/annotations/UnsafeRow.scala

```diff
@@ -254,15 +354,50 @@ class UnsafeRow(var ttBc: Broadcast[TypeTree],
 
   override def write(kryo: Kryo, output: Output) {
     kryo.writeObject(output, ttBc)
-    val enc = new Encoder(new DataOutputStream(output))
+    val enc = new Encoder()
     enc.writeRegionValue(ttBc.value.typ, region, offset)
+    kryo.writeObject(output, enc.outOff)
+    kryo.writeObject(output, enc.outMem)
+  }
+
+  private def writeObject(out: ObjectOutputStream) {
+    out.writeObject(ttBc)
+
+    val enc = new Encoder()
+    enc.writeRegionValue(ttBc.value.typ, region, offset)
+    assert(enc.outOff <= Int.MaxValue)
+    val smallOutOff = enc.outOff.toInt
+
+    out.writeInt(smallOutOff)
+    out.write(enc.outMem, 0, smallOutOff)
   }
 
   override def read(kryo: Kryo, input: Input) {
     val cTorrentBroadcast = Class.forName("org.apache.spark.broadcast.TorrentBroadcast")
     ttBc = kryo.readObject(input, cTorrentBroadcast).asInstanceOf[Broadcast[TypeTree]]
-    val dec = new Decoder(new DataInputStream(input))
+    val inOff = kryo.readObject[Long](input, classOf[Long])
```

Comment:
there's seriously no readLong in Kryo?

Reply:
Could you call `input.readLong()`?

Reply:
I think it's because he used writeObject above, which will write a java.lang.Long. If he switches to output.writeLong, then he can use input.readLong